lib/puppet/face/config.rb

```diff
@@ -163,7 +163,24 @@ def report_section_and_environment(section_name, environment_name)
       Puppet::FileSystem.touch(path)
       Puppet::FileSystem.open(path, nil, 'r+:UTF-8') do |file|
         Puppet::Settings::IniFile.update(file) do |config|
-          config.set(options[:section], name, value)
+          if options[:section] == "master"
+            # delete requested master section if it exists,
+            # as server section should be used
+            setting_string = config.delete("master", name)
+            if setting_string
+
+              if Puppet::Util::Log.sendlevel?(:info)
+                report_section_and_environment(options[:section], Puppet.settings[:environment])
+              end
+
+              puts(_("Deleted setting from '%{section_name}': '%{setting_string}', and adding it to 'server' section") %
```

Comment:
Currently if we have a value set in the `[master]` section and try to update it, the log shows the previous value which can be a bit misleading?

```sh-session
> cat ~/.puppetlabs/etc/puppet/puppet.conf
[master]
masterport = 8140
> bundle exec puppet config set masterport 8232 --section master
Deleted setting from 'master': 'masterport = 8140', and adding it to 'server' section
> cat ~/.puppetlabs/etc/puppet/puppet.conf
[master]
[server]
masterport = 8232
```
Can we only show the setting name instead of the entire value in this case? Or show both the old value and the new value.